modules/apps/portal-template/portal-template-soy-impl/src/main/java/com/liferay/portal/template/soy/internal/SoyManager.java

```diff
@@ -75,7 +76,7 @@ public void init() {
 	@Reference(unbind = "-")
 	public void setSingleVMPool(SingleVMPool singleVMPool) {
 		_soyTofuCacheHandler = new SoyTofuCacheHandler(
-			(PortalCache<HashSet<TemplateResource>, SoyTofuCacheBag>)
+			(PortalCache<Serializable, SoyTofuCacheBag>)
```

Comment:
So the reason you use Serializable rather than List<TemplateResource> here is to avoid "massive hashing"?